### ISSUE TITLE
Enhancement: Plan a trip from/to a stop

### ISSUE THREAD

**Issue by themonki** [OneBusAway/onebusaway-android#637]
I do not know how likely or good is this feature once you begin to stabilize the Trip Planner.

It would be good that the user can plan a trip by selecting a stop on the map or search for a stop, and using for default the current location.

Examples of the case is to visit someone else and this tells you the code of the nearest stop. It would only look for it and plan the trip.

**Comment by barbeau** [comment]
@themonki Agreed, this would be great.  I've been meaning to open this same issue myself.

A simple way to do this is to add a new option "Plan trip to/from this stop" to the ArrivalsListFragment overflow menu.  I think always populating the origin field with the location is probably ok, as the user can just choose "Reverse" from the menu to change it to the destination.

I'd also like to look at support launching the TripPlanActivity from an Intent using the standard `geo:latitude,longitude?z=zoom` format (see https://developers.google.com/maps/documentation/android-api/intents#uri_encoded_query_strings), but that's probably a separate issue (although handled in the same area of code).

**Comment by barbeau** [comment]
Requested by another user:

>Unless I missed this functionality, already being there, it would be very handy to be able to route the trip from where you currently are to the bus stop which you select, the stop one is viewing arrival times for.
For example, I select a stop by tapping its dot, I then decide that is the stop that I want to wait for the bus at.  I then tap my suggested button, "Take Me There".
This is useful if you are sitting somewhere, not the/a bus stop trying to determine what bus to take and you don't know the area.  The "Take Me There", maps one from where they are sitting to the bus stop they are viewing the arrival times for.
